internal/app/networkd/pkg/networkd/networkd.go

```diff
@@ -105,6 +107,20 @@ func New(config runtime.Configurator) (*Networkd, error) {
 			if strings.HasPrefix(device.Name, "bond") {
 				netconf[device.Name] = append(netconf[device.Name], nic.WithIgnore())
 			}
+
+			// Ignore links that do not have a carrier
+			carrier, err := ioutil.ReadFile("/sys/class/net/" + device.Name + "/carrier")
+			if err != nil {
+				result = multierror.Append(result, err)
+				continue
+			}
+
+			// note -- since we're only applying this to discovered interfaces
+			// that we don't have an explicit user defined, it wont negatively
+			// impact bond interfaces
+			if string(carrier) == "0" {
+				netconf[device.Name] = append(netconf[device.Name], nic.WithIgnore())
```

Comment:
being totally dumb on networking config, but what if carrier appears later after the initial boot?

someone powers on the switch or fixes the cable?